apps/discovery_api/lib/discovery_api_web/controllers/visualization_controller.ex

```diff
@@ -59,4 +65,13 @@ defmodule DiscoveryApiWeb.VisualizationController do
   defp under_visualizations_limit?(visualizations) do
     Enum.count(visualizations) <= Application.get_env(:discovery_api, :user_visualization_limit)
   end
+
+  defp get_allowed_actions(_visualization, nil), do: []
```

Comment:
Is there any more appropriate layer for this kind of thing than here? I think it's probably fine, and I'm not sure if it really belongs in the BL layer either.

Reply:
Good call out.  I'm thinking we wait and see how much more complicated this gets.  We could move it into its own plug or a helper module, but probably wouldn't do so until we have more rules.